lib/src/gotrue_client.dart

```diff
@@ -431,25 +431,28 @@ class GoTrueClient {
     final expiresAt = session.expiresAt;
 
     if (autoRefreshToken && expiresAt != null) {
-      if (_refreshTokenTimer != null) {
-        _refreshTokenTimer!.cancel();
-      }
+      _refreshTokenTimer?.cancel();
 
       final timeNow = (DateTime.now().millisecondsSinceEpoch / 1000).round();
       final expiresIn = expiresAt - timeNow;
       final refreshDurationBeforeExpires = expiresIn > 60 ? 60 : 1;
       final nextDuration = expiresIn - refreshDurationBeforeExpires;
       if (nextDuration > 0) {
         final timerDuration = Duration(seconds: nextDuration);
-        _refreshTokenTimer = Timer(timerDuration, () {
-          _callRefreshToken();
-        });
+        _setTokenRefreshTimer(timerDuration);
       } else {
         _callRefreshToken();
       }
     }
   }
 
+  void _setTokenRefreshTimer(Duration timerDuration) {
+    _refreshTokenTimer?.cancel();
```

Comment:
I have added a `cancel()` call before every time we set a new timer. 

About the stack overflow error, correct me if I am wrong, but as long as we are canceling the timer on every retry, I feel like stack overflow error would not occur and therefore we don't need a limit on how many retries we should attempt, but is my understanding correct?

Reply:
Yes, stack overflow would not happen in this case. Altough, keep trying and failing multiple times would be tough